### PR TITLE
feature/#1/initial_configuration

### PR DESCRIPTION
**Pull request by Segmentree**
# A little description about PR

Initial setup of Rust project and environment configuration. Setting the main tools and the basic to use unit tests.

## List of changes :

- [x] Create a new Rust app
- [x] Add .gitignore and exclude /target folder
- [x] VS code workspace configuration file
- [x] Setup unit tests
- [x] Add bubble-sort algorithm
- [x] Simple unit test for bubble-sort::sort function

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mod algorithms;
+fn main() {
```
Comment (rbruno95) [inline src/main.rs:2]: @Segmentree could you put an empty line before the main?

File: tests/bubble_sort.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+use rusthms::algorithms::bubble_sort::sort;
+
+#[test]
+fn it_sort_array() {
+    //declaring unsorting array of len 5
```
Comment (rbruno95) [inline tests/bubble_sort.rs:5]: @Segmentree unsorted, not unsorting

**Comment by rbruno95** [comment]
@Segmentree I think we don't need the main.rs file

File: src/algorithms/bubble_sort.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+pub fn sort(arr: &mut [i32]) {
+    let mut i = 0;
+    let mut j = 0;
+    let mut temp;
+    while i < arr.len() {
+        while j < arr.len() - 1 {
+            if arr[j] > arr[j + 1] {
+                temp = arr[j];
+                arr[j] = arr[j + 1];
+                arr[j + 1] = temp;
```
Comment (rbruno95) [inline src/algorithms/bubble_sort.rs:11]: @Segmentree check this out this link https://doc.rust-lang.org/stable/std/mem/fn.swap.html to be able to swap in one line

**Comment by rbruno95** [comment]
@Segmentree you can rewrite the bubble sort code using for loops to reduce the code to the half

File: src/algorithms/bubble_sort.rs (hunk 3 above)
Comment (Segmentree) [reply src/algorithms/bubble_sort.rs:11]: @rbruno95 seems to have a problem when you attempt to swap to array elements.

https://stackoverflow.com/questions/28294735/how-to-swap-the-elements-of-an-array-slice-or-vec

Comment (Segmentree) [reply src/algorithms/bubble_sort.rs:11]: @rbruno95 forget about it you can call the swap function in the array class and pass two indexes.

**Comment by rbruno95** [comment]
> @Segmentree I think we don't need the main.rs file
> 
> @rbruno95 maybe we can use it to create an algorithm playground where you select what algorithm you want to use and pass to it whatever case you want.

@Segmentree makes sense

**Review by rbruno95 — APPROVED** [review]

**Pull request merged**